flink-docs/src/main/java/org/apache/flink/docs/rest/OpenApiSpecGenerator.java

```diff
@@ -452,7 +447,26 @@ private static Content createContentWithMediaType(String mediaType, Schema schem
     }
 
     private static Schema<?> getSchema(Type type) {
-        return modelConverterContext.resolve(new AnnotatedType(type).resolveAsRef(true));
+        final AnnotatedType annotatedType = new AnnotatedType(type).resolveAsRef(true);
+        final Schema<?> schema = modelConverterContext.resolve(annotatedType);
+        if (type instanceof Class<?>) {
+            final Class<?> clazz = (Class<?>) type;
+            ApiGeneratorUtils.findAdditionalFieldType(clazz)
+                    .map(OpenApiSpecGenerator::getSchema)
+                    .ifPresent(
+                            additionalPropertiesSchema -> {
+                                // We can not reuse the schema instance returned by the converter
+                                // above, because it's a different instance than the resolver has
+                                // defined, so we wouldn't see the changes later when populating
+                                // OpenAPI components.
+                                final Schema<?> definedModel =
```

Comment:
Shouldn't `definedModel` be returned instead of the resolved one?
(I guess not, just want to double-check)

Reply:
Thanks for clarifying and updating the comment, it makes it much easier to understand now!